src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -398,6 +443,16 @@ public CSharpCompilationOptions WithConcurrentBuild(bool concurrentBuild)
             return new CSharpCompilationOptions(this) { ConcurrentBuild = concurrentBuild };
         }
 
+        public CSharpCompilationOptions WithDeterminism(bool deterministic)
+        {
```

Comment:
Should this be called "WithDeterministic" ?   It seems to refer to a compiler flag to not to an overral concept.